### PR TITLE
Adds `eddsa-jcs-2022` tests.

### PR DESCRIPTION
**Pull request by JSAssassin**
TODO:

- [x] Add Changelog entry
- [x] Update `README.md`

### CODE REVIEW

File: tests/40-jcs-create.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
+/*!
+ * Copyright 2023 Digital Bazaar, Inc. All Rights Reserved
```
Comment (TallTed) [inline tests/40-jcs-create.js:2]: again, `copyright W3C CC-whatever` also/instead

File: tests/10-rdfc-create.js [hunk 2]
```diff
@@ -1,7 +1,6 @@
 /*!
  * Copyright 2022-2023 Digital Bazaar, Inc. All Rights Reserved
```
Comment (TallTed) [inline tests/10-rdfc-create.js:2]: again, `copyright W3C CC-whatever` also/instead

File: tests/50-jcs-verify.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
+/*!
+ * Copyright (c) 2023 Digital Bazaar, Inc. All rights reserved.
```
Comment (TallTed) [inline tests/50-jcs-verify.js:2]: again, `copyright W3C CC-whatever` also/instead

File: tests/60-jcs-interop.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright (c) 2023 Digital Bazaar, Inc. All rights reserved.
```
Comment (TallTed) [inline tests/60-jcs-interop.js:2]: again, `copyright W3C CC-whatever` also/instead

File: tests/test-config.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+/*!
+ * Copyright 2023 Digital Bazaar, Inc. All Rights Reserved
```
Comment (TallTed) [inline tests/test-config.js:2]: again, `copyright W3C CC-whatever` also/instead

**Review by TallTed — CHANGES_REQUESTED** [review]

File: tests/40-jcs-create.js (hunk 1 above)
Comment (JSAssassin) [reply tests/40-jcs-create.js]: Added `LICENSE.md` and updated the copyright headers here - [2f768fa](https://github.com/w3c/vc-di-eddsa-test-suite/pull/32/commits/2f768fabbb79539cad13543662844166785420fb)

**Review by BigBlueHat — APPROVED** [review]
Excellent work, as ever, @JSAssassin. Thank you!

It's great to see the licensing stuff in place also. Appreciate it!

File: tests/40-jcs-create.js [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+            const publicKeyMultibaseBytes = bs58Decode({
+              id: publicKeyMultibase
+            });
+            publicKeyMultibaseBytes.byteLength.should.equal(34, 'Expected ' +
+              '"publicKeyMultibase" value of the verification method to ' +
+              'be 34 bytes in length.');
+          }
+        });
+        it('"proofValue" field when decoded to raw bytes, MUST be 64 bytes ' +
+          'in length if the associated public key is 32 bytes or 114 bytes ' +
```
Comment (TallTed) [inline tests/40-jcs-create.js:132]:
```suggestion
          'in length if the associated public key is 32 bytes, or 114 bytes ' +
```

File: tests/40-jcs-create.js [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+            const vmBytes = await getPublicKeyBytes({
+              did: proof.verificationMethod});
+            vmBytes.byteLength.should.be.oneOf([32, 57], 'Expected public ' +
+              'key bytes to be either 32 or 57 bytes.');
+            if(vmBytes.byteLength === 32) {
+              valueBytes.byteLength.should.equal(64, 'Expected 64 bytes ' +
+                'proofValue for 32 bytes key.');
+            } else {
+              valueBytes.byteLength.should.equal(114, 'Expected 114 bytes ' +
+                'proofValue for 57 bytes key.');
```
Comment (TallTed) [inline tests/40-jcs-create.js:156]:
```suggestion
              valueBytes.byteLength.should.equal(64, 'Expected 64 byte ' +
                'proofValue for 32 byte key.');
            } else {
              valueBytes.byteLength.should.equal(114, 'Expected 114 byte ' +
                'proofValue for 57 byte key.');
```

File: tests/40-jcs-create.js [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+            }
+          }
+        });
+        it('"proof" MUST verify when using a conformant verifier.',
+          async function() {
+            this.test.cell = {columnId, rowId: this.test.title};
+            should.exist(verifier, 'Expected implementation to have a VC ' +
+              'HTTP API compatible verifier.');
```
Comment (TallTed) [inline tests/40-jcs-create.js:164]:
```suggestion
              'API compatible verifier.');
```

**Review by TallTed — CHANGES_REQUESTED** [review]
minor, human facing

**Review by BigBlueHat — APPROVED** [review]
Looks good! If the `it()` topics are verbatim text from specs, I'd suggest _not_ changing them. If they are not, then I support @TallTed's suggestions (generally...though I'm not sure about that comma suggestion...). 😉

File: LICENSE.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+BSD 3-Clause License
+
+Copyright (c) 2024, W3C Verifiable Credentials Working Group Contributors
```
[the remark this answers is not in the document]
Comment (JSAssassin) [reply LICENSE.md]: Updated [7370318](https://github.com/w3c/vc-di-eddsa-test-suite/pull/32/commits/7370318e4f46d841aa092c3f9f8a7b1e195dc52b)

File: tests/40-jcs-create.js (hunk 6 above)
Comment (JSAssassin) [reply tests/40-jcs-create.js:132]: This is a verbatim text from the specs and the test description will change in future, so leaving this as it is for now.

File: tests/40-jcs-create.js (hunk 7 above)
Comment (JSAssassin) [reply tests/40-jcs-create.js]: Thanks, fixed [18a9bcb](https://github.com/w3c/vc-di-eddsa-test-suite/pull/32/commits/18a9bcbe80583fe8f8d4a96df9c55c0b3897ced6)

File: tests/40-jcs-create.js (hunk 8 above)
Comment (JSAssassin) [reply tests/40-jcs-create.js]: Updated [e0739b2](https://github.com/w3c/vc-di-eddsa-test-suite/pull/32/commits/e0739b2677a41f25a4997da39e5482db066a86f8)

**Comment by TallTed** [comment]
> [@BigBlueHat] If the it() topics are verbatim text from specs, I'd suggest not changing them. If they are not, then I support @TallTed's suggestions

Fair point! Usually they are not verbatim, but if they happen to be here, I agree with you.

> [@BigBlueHat]  (generally...though I'm not sure about that comma suggestion...)

That comma is optional according to English guidelines. I believe it will help with comprehension of this text, such that it need not be read a few times to be sure its understood.

File: tests/10-rdfc-create.js [hunk 10]
```diff
@@ -151,18 +150,18 @@ describe('eddsa-rdfc-2022 (create)', function() {
               'key bytes to be either 32 or 57 bytes.');
             if(vmBytes.byteLength === 32) {
               valueBytes.byteLength.should.equal(64, 'Expected 64 bytes ' +
```
Comment (TallTed) [inline tests/10-rdfc-create.js:152]:
```suggestion
              valueBytes.byteLength.should.equal(64, 'Expected 64 byte ' +
```

File: tests/10-rdfc-create.js [hunk 11]
```diff
@@ -151,18 +150,18 @@ describe('eddsa-rdfc-2022 (create)', function() {
...
-                'proofValue for 32 bytes key.');
+                'proofValue for 32 byte key.');
             } else {
               valueBytes.byteLength.should.equal(114, 'Expected 114 bytes ' +
```
Comment (TallTed) [inline tests/10-rdfc-create.js:155]:
```suggestion
              valueBytes.byteLength.should.equal(114, 'Expected 114 byte ' +
```

File: tests/40-jcs-create.js [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+            'eddsa-jcs-2022 cryptosuite.');
+          for(const proof of eddsa2022Proofs) {
+            should.exist(proof.proofValue, 'Expected a proof value on ' +
+              'the proof.');
+            const valueBytes = bs58Decode({id: proof.proofValue});
+            should.exist(proof.verificationMethod);
+            const vmBytes = await getPublicKeyBytes({
+              did: proof.verificationMethod});
+            vmBytes.byteLength.should.be.oneOf([32, 57], 'Expected public ' +
+              'key bytes to be either 32 or 57 bytes.');
```
Comment (TallTed) [inline tests/40-jcs-create.js:150]:
```suggestion
              'key length to be either 32 or 57 bytes.');
```

File: tests/40-jcs-create.js [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+            if(vmBytes.byteLength === 32) {
+              valueBytes.byteLength.should.equal(64, 'Expected 64 bytes ' +
```
Comment (TallTed) [inline tests/40-jcs-create.js:152]:
```suggestion
              valueBytes.byteLength.should.equal(64, 'Expected 64 byte ' +
```

File: tests/40-jcs-create.js [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+                'proofValue for 32 byte key.');
+            } else {
+              valueBytes.byteLength.should.equal(114, 'Expected 114 bytes ' +
```
Comment (TallTed) [inline tests/40-jcs-create.js:155]:
```suggestion
              valueBytes.byteLength.should.equal(114, 'Expected 114 byte ' +
```

File: tests/10-rdfc-create.js [hunk 15]
```diff
@@ -151,18 +150,18 @@ describe('eddsa-rdfc-2022 (create)', function() {
               'key bytes to be either 32 or 57 bytes.');
```
Comment (TallTed) [inline tests/10-rdfc-create.js:150]:
```suggestion
              'key length to be either 32 or 57 bytes.');
```

**Review by TallTed — CHANGES_REQUESTED** [review]
If any of these `bytes`->`byte` changes (that I'm re-requesting) look odd, try copying and pasting the full message, so you read the whole thing at once. (If you didn't intentionally reject these, note that this is why I suggest using the "commit" buttons in the GitHub interface, rather than manually re-entering my requested changes. Manual re-entry is prone to errors which are not limited to omitting part of a suggetsion.)

**Comment by JSAssassin** [comment]
@TallTed Sorry. I didn't reject the suggestions intentionally, it was an oversight on my part. I didn't use the "commit" buttons on the GitHub interface previously to avoid making the same commit message for similar suggestions. Anyways, I've now used the "commit" buttons, so all your suggestions should be addressed. Could you please review and approve if it looks good to you. Thank you.

**Comment by aljones15** [comment]
@TallTed is this PR (vc-di-eddsa-test-suite/pulls/32) ready for merge now or do you have more issues with it?

**Review by TallTed — APPROVED** [review]

**Comment by aljones15** [comment]
@TallTed thanks a ton for the review and the eddsa-jcs-tests have been merged now.

**Pull request merged**